catalog/pages/inputs/CheckBoxExample.js

```diff
@@ -0,0 +1,120 @@
+// this components was created for checkboxes demonstration purpose
+import React, { Component, Fragment } from "react";
+import PropTypes from "prop-types";
+import TOGGLE_SIZES from "../../../src/components/Input/Toggle/constants";
+import { CheckBoxButton } from "../../../src/components/Input";
+
+export default class CheckboxWrapperExample extends Component {
+  static propTypes = {
+    size: PropTypes.oneOf(TOGGLE_SIZES).isRequired
+  };
+
+  state = {
+    fruits: false,
+    subOptions: {
+      apple: false,
+      peach: false,
+      banana: false
+    }
+  };
+
+  handleChange = e => {
+    let fruits = false;
+    const { subOptions } = this.state;
+    const { name, checked } = e.target;
+
+    if (name === "fruits") {
+      fruits = checked;
+      Object.keys(this.state.subOptions).forEach(el => {
+        subOptions[el] = fruits;
```

Comment:
Please do not modify the state via side-effects. It would be better to allocate new memory for the purpose of deriving the new state from the user's change

Reply:
Right. Thank you. Fixed now.